pkg/controllers/nodetemplate/suite_test.go

```diff
@@ -257,92 +257,100 @@ var _ = Describe("AWSNodeTemplateController", func() {
-				return securitygroup.ID
-			})
-			Expect(securityGroupsIDInStatus).To(Equal(securityGroupsIDs))
+			Expect(nodeTemplate.Status.SecurityGroups).To(BeNil())
 		})
 		It("Should update AWSNodeTemplate status for Security Groups", func() {
 			ExpectApplied(ctx, env.Client, nodeTemplate)
 			ExpectReconcileSucceeded(ctx, controller, client.ObjectKeyFromObject(nodeTemplate))
 			nodeTemplate = ExpectExists(ctx, env.Client, nodeTemplate)
-			securityGroupsIDs, _ := awsEnv.SecurityGroupProvider.List(ctx, nodeTemplate)
-			securityGroupsIDInStatus := lo.Map(nodeTemplate.Status.SecurityGroups, func(securitygroup v1alpha1.SecurityGroup, _ int) string {
-				return securitygroup.ID
+			securityGroups, _ := awsEnv.SecurityGroupProvider.List(ctx, nodeTemplate)
+			correctSecurityGroups := lo.Map(securityGroups, func(securityGroup *ec2.SecurityGroup, _ int) v1alpha1.SecurityGroup {
+				return v1alpha1.SecurityGroup{
+					ID:   *securityGroup.GroupId,
+					Name: *securityGroup.GroupName,
+				}
 			})
-			Expect(securityGroupsIDInStatus).To(Equal(securityGroupsIDs))
+			Expect(nodeTemplate.Status.SecurityGroups).To(Equal(correctSecurityGroups))
 		})
 		It("Should resolve a valid selectors for Security Groups by tags", func() {
 			nodeTemplate.Spec.SecurityGroupSelector = map[string]string{`Name`: `test-security-group-1,test-security-group-2`}
 			ExpectApplied(ctx, env.Client, nodeTemplate)
 			ExpectReconcileSucceeded(ctx, controller, client.ObjectKeyFromObject(nodeTemplate))
 			nodeTemplate = ExpectExists(ctx, env.Client, nodeTemplate)
-			securityGroupsIDs, _ := awsEnv.SecurityGroupProvider.List(ctx, nodeTemplate)
-			correctSecurityGroupsIDs := lo.Map(securityGroupsIDs, func(securitygroup string, _ int) v1alpha1.SecurityGroup {
+			securityGroups, _ := awsEnv.SecurityGroupProvider.List(ctx, nodeTemplate)
+			correctSecurityGroups := lo.Map(securityGroups, func(securitygroup *ec2.SecurityGroup, _ int) v1alpha1.SecurityGroup {
 				return v1alpha1.SecurityGroup{
-					ID: securitygroup,
+					ID:   *securitygroup.GroupId,
+					Name: *securitygroup.GroupName,
 				}
 			})
-			Expect(nodeTemplate.Status.SecurityGroups).To(Equal(correctSecurityGroupsIDs))
+			Expect(nodeTemplate.Status.SecurityGroups).To(Equal(correctSecurityGroups))
 		})
 		It("Should resolve a valid selectors for Security Groups by ids", func() {
 			nodeTemplate.Spec.SecurityGroupSelector = map[string]string{`aws-ids`: `sg-test1`}
 			ExpectApplied(ctx, env.Client, nodeTemplate)
 			ExpectReconcileSucceeded(ctx, controller, client.ObjectKeyFromObject(nodeTemplate))
 			nodeTemplate = ExpectExists(ctx, env.Client, nodeTemplate)
-			securityGroupsIDs, _ := awsEnv.SecurityGroupProvider.List(ctx, nodeTemplate)
-			correctSecurityGroupsIDs := lo.Map(securityGroupsIDs, func(securitygroup string, _ int) v1alpha1.SecurityGroup {
+			securityGroups, _ := awsEnv.SecurityGroupProvider.List(ctx, nodeTemplate)
```

Comment:
You probably want to ensure the static values here rather than dynamically checking these values. Otherwise, you risk a bug in the List call not being caught

Reply:
I will make a separate PR for the subnets status testing to make similar changes.